spec/support/controller_macros.rb

```diff
@@ -0,0 +1,16 @@
+module ControllerMacros
+  # def login_admin
+  #   before(:each) do
+  #     @request.env["devise.mapping"] = Devise.mappings[:admin]
+  #     sign_in FactoryBot.create(:admin) # Using factory bot as an example
+  #   end
+  # end
+
+  def login_user
```

Comment:
This is very nice. Would have expected the namespace to be more like `/spec/helpers/login_helper` or something like that though maybe?

Reply:
I was following the advice in this article https://yuta-san.medium.com/a-simple-login-test-with-rspec-devise-factorybot-in-rails-29aeb2ebc4ab which I think is based on the information contained in here https://github.com/heartcombo/devise/wiki/How-To:-Test-controllers-with-Rails-%28and-RSpec%29

Reply:
I'm happy to change this if you think it should be changed 😄